Gems/Atom/RPI/Code/Source/RPI.Reflect/Image/StreamingImageAsset.cpp

```diff
@@ -93,18 +93,50 @@ namespace AZ
         {
             return m_totalImageDataSize;
         }
-        
+
+        RHI::ImageDescriptor StreamingImageAsset::GetImageDescriptorForMipLevel(AZ::u32 mipLevel) const
+        {
+            RHI::ImageDescriptor imageDescriptor = GetImageDescriptor();
+
+            const ImageMipChainAsset* mipChainAsset = GetImageMipChainAsset(mipLevel);
+            if (mipChainAsset)
+            {
+                auto mipChainIndex = GetMipChainIndex(mipLevel);
+                auto mipChainOffset = aznumeric_cast<AZ::u32>(GetMipLevel(mipChainIndex));
+                auto layout = mipChainAsset->GetSubImageLayout(mipLevel - mipChainOffset);
+
+                imageDescriptor.m_size = layout.m_size;
+            }
```

Comment:
What should happen in the 'else' case here? Right now, it will return a valid descriptor with the top mip level's size, which doesn't seem right.